crates/swc_ecma_transforms_compat/tests/es2015_regenerator.rs

```diff
@@ -1169,6 +1169,311 @@ test_exec!(
+              case 0:
+                refreshThenRequest = function _refreshThenRequest1() {
+                  return _refreshThenRequest.apply(this, arguments);
+                };
+                _refreshThenRequest = function __refreshThenRequest() {
+                  _refreshThenRequest = _asyncToGenerator(
+                    regeneratorRuntime.mark(function _callee() {
+                      return regeneratorRuntime.wrap(function _callee$(_ctx) {
+                        while (1)
+                          switch (_ctx.prev = _ctx.next) {
+                            case 0:
+                            case "end":
+                              return _ctx.stop();
+                          }
+                      }, _callee);
+                    })
+                  );
+                  return _refreshThenRequest.apply(this, arguments);
+                };
+                _ctx1.next = 4;
+                return refreshThenRequest();
+              case 4:
+                return _ctx1.abrupt("return", true);
+              case 7:
+              case "end":
+                return _ctx1.stop();
+            }
+        }, _callee1);
+      })
+    );
+    return _pureRequester.apply(this, arguments);
+  }
+}
+"#
+);
+
+test_exec!(
+    Syntax::default(),
+    |_| chain!(async_to_generator(), tr(())),
+    hoist_function_in_async_issue_2556_5,
+    r#"
+function requester() {
+    return pureRequester;
+  
+    async function pureRequester() {
+      await refreshThenRequest();
+      return true;
+  
+      async function refreshThenRequest() {}
+    }
+  }
+  
+  return requester()().then(function test(result) {
+      expect(result).toBe(true);
+  });
+"#
+);
+
+test_exec!(
+    Syntax::default(),
+    |_| chain!(async_to_generator(), tr(())),
+    issue_2620,
+    r#"
+async function main() {
+    class Weird1 {
+        weird = async () => {
+            return !!this;
+        };
+
+        decoy1 = async () => {};
+        decoy2 = () => {};
+    }
+
+    class Weird2 {
+        weird = async () => {
+            return !!this;
+        };
+
+        //    decoy1 = async () => { };
+        decoy2 = () => {};
+    }
+
+    class Weird3 {
+        weird = async () => {
+            return !!this;
+        };
+
+        decoy1 = async () => {};
+        //    decoy2 = () => { };
+    }
+
+    class Weird4 {
+        decoy1 = async () => {};
+        decoy2 = () => {};
+
+        weird = async () => {
+            return !!this;
+        };
+    }
+
+    return Promise.all([
+        new Weird4().weird(),
+        new Weird3().weird(),
+        new Weird2().weird(),
+        new Weird1().weird(),
+    ]);
+}
+
+return main().then((results) => {
+    expect(results).toEqual([true, true, true, true]);
+});
+"#
+);
+
+test!(
+    Syntax::default(),
+    |_| chain!(async_to_generator(), tr(())),
+    hoist_function_in_async_issue_2556_6,
+    r#"
+async function foo(a) {
+    return bar1;
+
+    const bar1 = "bar1";
```

Comment:
@kdy1 Same test code here.
Dead codes are correctly removed.

I cannot to figure out the difference between test files and inline test code.

Reply:
I think the difference is `resolver`.